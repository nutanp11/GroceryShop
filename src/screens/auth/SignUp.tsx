import {
  StyleSheet,
  Text,
  View,
  ScrollView,
  TextStyle,
  ViewStyle,
  Alert,
} from 'react-native';
import React, { useState } from 'react';
import { Colors } from '../../constants/Colors';
import { Fonts } from '../../constants/Fonts';
import AppButton from '../../components/AppButton';
import TextInputComp from '../../components/TextInputComp';
import HeaderText from '../../components/HeaderText';
import CheckBox from '@react-native-community/checkbox';
import PhoneInput from 'react-native-phone-input';
import { ALREADY_ACCOUNT } from '../../constants/Strings';

type SignUpNavigationProp = {
  navigate: (screen: string) => void;
};

type SignUpProps = {
  navigation: SignUpNavigationProp;
};

const SignUp: React.FC<SignUpProps> = ({ navigation }) => {
  const [name, setName] = useState<string>('');
  const [email, setEmail] = useState<string>('');
  const [password, setPassword] = useState<string>('');
  const [phone_number, setPhoneNumber] = useState<string>('');
  const [toggleCheckBox, setToggleCheckBox] = useState<boolean>(false);

  const validateForm = (): boolean => {
    if (!name.trim()) {
      Alert.alert('Validation Failed', 'Please enter your full name.');
      return false;
    }
    if (!email.trim()) {
      Alert.alert('Validation Failed', 'Please enter your email address.');
      return false;
    }
    // Simple email regex validation
    const emailRegex = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,6}$/;
    if (!emailRegex.test(email)) {
      Alert.alert('Validation Failed', 'Please enter a valid email address.');
      return false;
    }
    if (!password.trim()) {
      Alert.alert('Validation Failed', 'Please enter your password.');
      return false;
    }
    if (password.length < 6) {
      Alert.alert('Validation Failed', 'Password should be at least 6 characters.');
      return false;
    }
    if (!phone_number.trim()) {
      Alert.alert('Validation Failed', 'Please enter your phone number.');
      return false;
    }
    if (!toggleCheckBox) {
      Alert.alert('Validation Failed', 'You must agree to the terms and conditions.');
      return false;
    }
    return true;
  };

  const onPressCreateAccount = () => {
    if (validateForm()) {
      Alert.alert('Success', 'Your account has been created.');
      navigation.navigate('SignIn');
    }
  };

  return (
    <ScrollView
      keyboardShouldPersistTaps="always"
      contentContainerStyle={{ flexGrow: 1 }}>
      <View style={styles.container}>
        <View style={styles.topview}>
          <HeaderText title={'Basket'} />
          <Text style={styles.loginText}>Create an Account</Text>
          <Text style={styles.subtext}>Please type your information</Text>

          <TextInputComp
            value={name}
            placeholderText={'Full name'}
            onChangeText={(text: string) => setName(text)}
          />
          <TextInputComp
            value={email}
            placeholderText={'Email address'}
            onChangeText={(text: string) => setEmail(text)}
          />
          <TextInputComp
            value={password}
            placeholderText={'Password'}
            onChangeText={(text: string) => setPassword(text)}
          />

          <View style={styles.phoneContainer}>
            <View style={styles.phoneInputWrapper}>
              <PhoneInput
                initialCountry={'us'}
                initialValue="13178675309"
                textProps={{
                  placeholder: 'Enter a phone number...',
                }}
              />
            </View>

            <TextInputComp
              value={phone_number}
              placeholderText={'000-00-0000'}
              onChangeText={(text: string) => setPhoneNumber(text)}
              keyboardType="phone-pad"
            />
          </View>

          <View style={styles.checkBoxContainer}>
            <CheckBox
              disabled={false}
              value={toggleCheckBox}
              onValueChange={(newValue: boolean) => setToggleCheckBox(newValue)}
            />
            <Text style={styles.agreementText}>
              By joining, I agree to receive emails from Geeta.
            </Text>
          </View>
          <AppButton title={'Create Account'} onPress={onPressCreateAccount} />
        </View>

        <View style={styles.bottomview}>
          <View style={styles.signiInview}>
            <Text style={styles.graytext}>{ALREADY_ACCOUNT} </Text>
            <Text
              onPress={() => navigation.navigate('SignIn')}
              style={styles.blacktext}>
              Sign In
            </Text>
          </View>
        </View>
      </View>
    </ScrollView>
  );
};

export default SignUp;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.appBackground,
  } as ViewStyle,
  topview: {
    flex: 0.8,
    padding: 10,
  } as ViewStyle,
  signiInview: {
    flexDirection: 'row',
    alignSelf: 'center',
  } as ViewStyle,
  bottomview: {
    flex: 0.2,
    justifyContent: 'space-between',
  } as ViewStyle,
  loginText: {
    marginTop: 30,
    fontSize: 24,
    fontWeight: '900',
    fontFamily: Fonts.bold,
    alignSelf: 'center',
  } as TextStyle,
  subtext: {
    fontSize: 14,
    fontWeight: '600',
    lineHeight: 35,
    color: Colors.textGray,
    fontFamily: Fonts.regular,
    alignSelf: 'center',
  } as TextStyle,
  phoneContainer: {
    justifyContent: 'space-between',
    flexDirection: 'row',
    width: '90%',
    alignSelf: 'center',
  } as ViewStyle,
  phoneInputWrapper: {
    backgroundColor: Colors.grayBack,
    alignSelf: 'center',
    padding: 20,
  } as ViewStyle,
  checkBoxContainer: {
    justifyContent: 'space-between',
    flexDirection: 'row',
    width: '90%',
    alignSelf: 'center',
    marginVertical: 20,
  } as ViewStyle,
  agreementText: {
    fontFamily: Fonts.regular,
    alignSelf: 'center',
    color: Colors.textGray,
  } as TextStyle,
  graytext: {
    fontSize: 14,
    fontWeight: '600',
    color: Colors.textGray,
    fontFamily: Fonts.regular,
    alignSelf: 'center',
  } as TextStyle,
  blacktext: {
    fontSize: 14,
    fontFamily: Fonts.semibold,
  } as TextStyle,
  inpuStyle: { width: '75%' } as TextStyle,
});
