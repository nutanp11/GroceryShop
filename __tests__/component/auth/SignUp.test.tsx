import { fireEvent, render } from '@testing-library/react-native';
import {
    CREATE_ACC_BTN,
  CREATE_ACCOUNT,
  ENTER_EMAIL,
  FULL_NAME,
  LOG_IN,
  PASSWORD,
} from '../../../src/constants/Strings';
import { Alert } from 'react-native';
import SignUp from '../../../src/screens/auth/SignUp';

describe('SignUp Screen',()=> {
    it('should match snapshot', () => {
        const { toJSON } = render(<SignUp navigation={{ navigate: jest.fn() }} />);
        expect(toJSON()).toMatchSnapshot();
      });


      it('should render correctly', () => {
        const { getByPlaceholderText, getByText } = render(
          <SignUp navigation={{ navigate: jest.fn() }} />
        );
      
        expect(getByPlaceholderText(FULL_NAME)).toBeTruthy();
        expect(getByPlaceholderText(ENTER_EMAIL)).toBeTruthy();
        expect(getByPlaceholderText(PASSWORD)).toBeTruthy();
        expect(getByPlaceholderText('000-00-0000')).toBeTruthy();
        expect(getByText(CREATE_ACC_BTN)).toBeTruthy();
      });


      it('should handle input changes correctly', () => {
        const { getByPlaceholderText } = render(
          <SignUp navigation={{ navigate: jest.fn() }} />
        );
      
        const nameInput = getByPlaceholderText(FULL_NAME);
        const emailInput = getByPlaceholderText(ENTER_EMAIL);
        const passwordInput = getByPlaceholderText(PASSWORD);
        const phoneInput = getByPlaceholderText('000-00-0000');
      
        fireEvent.changeText(nameInput, 'John Doe');
        fireEvent.changeText(emailInput, 'john.doe@example.com');
        fireEvent.changeText(passwordInput, 'password123');
        fireEvent.changeText(phoneInput, '123-456-7890');
      
        expect(nameInput.props.value).toBe('John Doe');
        expect(emailInput.props.value).toBe('john.doe@example.com');
        expect(passwordInput.props.value).toBe('password123');
        expect(phoneInput.props.value).toBe('123-456-7890');
      });

})