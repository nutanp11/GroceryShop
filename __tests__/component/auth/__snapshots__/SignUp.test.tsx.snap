// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Screen should match snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "flexGrow": 1,
    }
  }
  keyboardShouldPersistTaps="always"
>
  <View>
    <View
      style={
        {
          "backgroundColor": "#ffffff",
          "flex": 1,
        }
      }
    >
      <View
        style={
          {
            "flex": 0.8,
            "padding": 10,
          }
        }
      >
        <View>
          <Text
            style={
              {
                "alignSelf": "center",
                "fontFamily": "Nunito-Bold",
                "fontSize": 54,
                "fontWeight": "900",
              }
            }
          >
            Basket
          </Text>
        </View>
        <Text
          style={
            {
              "alignSelf": "center",
              "fontFamily": "Nunito-Bold",
              "fontSize": 24,
              "fontWeight": "900",
              "marginTop": 30,
            }
          }
        >
          Create an Account
        </Text>
        <Text
          style={
            {
              "alignSelf": "center",
              "color": "#999999",
              "fontFamily": "Nunito-Regular",
              "fontSize": 14,
              "fontWeight": "600",
              "lineHeight": 35,
            }
          }
        >
          Please type your information
        </Text>
        <View
          style={
            [
              {
                "alignSelf": "center",
                "backgroundColor": "#F4F4F4",
                "borderRadius": 5,
                "marginVertical": 10,
                "padding": 5,
                "width": "90%",
              },
              undefined,
            ]
          }
        >
          <TextInput
            keyboardType="default"
            onChangeText={[Function]}
            placeholder="Full name"
            placeholderTextColor="#000000"
            style={
              {
                "color": "#000000",
                "fontFamily": "Nunito-Bold",
                "fontSize": 14,
                "fontWeight": "900",
                "padding": 15,
              }
            }
            value=""
          />
        </View>
        <View
          style={
            [
              {
                "alignSelf": "center",
                "backgroundColor": "#F4F4F4",
                "borderRadius": 5,
                "marginVertical": 10,
                "padding": 5,
                "width": "90%",
              },
              undefined,
            ]
          }
        >
          <TextInput
            keyboardType="default"
            onChangeText={[Function]}
            placeholder="Email address"
            placeholderTextColor="#000000"
            style={
              {
                "color": "#000000",
                "fontFamily": "Nunito-Bold",
                "fontSize": 14,
                "fontWeight": "900",
                "padding": 15,
              }
            }
            value=""
          />
        </View>
        <View
          style={
            [
              {
                "alignSelf": "center",
                "backgroundColor": "#F4F4F4",
                "borderRadius": 5,
                "marginVertical": 10,
                "padding": 5,
                "width": "90%",
              },
              undefined,
            ]
          }
        >
          <TextInput
            keyboardType="default"
            onChangeText={[Function]}
            placeholder="Password"
            placeholderTextColor="#000000"
            style={
              {
                "color": "#000000",
                "fontFamily": "Nunito-Bold",
                "fontSize": 14,
                "fontWeight": "900",
                "padding": 15,
              }
            }
            value=""
          />
        </View>
        <View
          style={
            {
              "alignSelf": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "90%",
            }
          }
        >
          <View
            style={
              {
                "alignSelf": "center",
                "backgroundColor": "#F4F4F4",
                "padding": 20,
              }
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <View
                accessibilityLabel="United States"
                accessibilityRole="imagebutton"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <Image
                  accessibilityIgnoresInvertColors={true}
                  source={
                    {
                      "testUri": "../../react-native-phone-input/dist/resources/flags/images/us.png",
                    }
                  }
                  style={
                    [
                      {
                        "backgroundColor": "#cecece",
                        "borderColor": "#cecece",
                        "borderRadius": 2,
                        "borderWidth": 0.5,
                        "height": 20,
                        "width": 30,
                      },
                      undefined,
                    ]
                  }
                />
              </View>
              <View
                style={
                  {
                    "flex": 1,
                    "marginLeft": 10,
                  }
                }
              >
                <TextInput
                  accessibilityLabel="Telephone input"
                  autoCorrect={false}
                  editable={true}
                  keyboardType="phone-pad"
                  onChangeText={[Function]}
                  placeholder="Enter a phone number..."
                  style={
                    [
                      {
                        "height": 20,
                        "justifyContent": "center",
                        "padding": 0,
                      },
                      undefined,
                    ]
                  }
                  underlineColorAndroid="rgba(0,0,0,0)"
                  value="+13178675309"
                />
              </View>
            </View>
          </View>
          <View
            style={
              [
                {
                  "alignSelf": "center",
                  "backgroundColor": "#F4F4F4",
                  "borderRadius": 5,
                  "marginVertical": 10,
                  "padding": 5,
                  "width": "90%",
                },
                undefined,
              ]
            }
          >
            <TextInput
              keyboardType="phone-pad"
              onChangeText={[Function]}
              placeholder="000-00-0000"
              placeholderTextColor="#000000"
              style={
                {
                  "color": "#000000",
                  "fontFamily": "Nunito-Bold",
                  "fontSize": 14,
                  "fontWeight": "900",
                  "padding": 15,
                }
              }
              value=""
            />
          </View>
        </View>
        <View
          style={
            {
              "alignSelf": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginVertical": 20,
              "width": "90%",
            }
          }
        >
          <View
            accessibilityRole="checkbox"
            accessibilityState={
              {
                "checked": false,
                "disabled": false,
              }
            }
            accessibilityValue={
              {
                "text": "off",
              }
            }
            accessible={true}
            pointerEvents="auto"
          >
            <RNCCheckbox
              forwardedRef={null}
              onValueChange={[Function]}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "height": 32,
                    "width": 32,
                  },
                  undefined,
                ]
              }
              value={false}
            />
          </View>
          <Text
            style={
              {
                "alignSelf": "center",
                "color": "#999999",
                "fontFamily": "Nunito-Regular",
              }
            }
          >
            By joining, I agree to receive emails from Geeta.
          </Text>
        </View>
        <View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#2F2E41",
                "borderRadius": 10,
                "opacity": 1,
                "padding": 10,
                "width": "80%",
              }
            }
          >
            <Text
              style={
                {
                  "alignSelf": "center",
                  "color": "#FFFFFF",
                  "fontFamily": "Nunito-Bold",
                  "fontSize": 24,
                  "fontWeight": "900",
                  "lineHeight": 45,
                }
              }
            >
              Create Account
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "flex": 0.2,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            {
              "alignSelf": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "alignSelf": "center",
                "color": "#999999",
                "fontFamily": "Nunito-Regular",
                "fontSize": 14,
                "fontWeight": "600",
              }
            }
          >
            Already have an account?
             
          </Text>
          <Text
            onPress={[Function]}
            style={
              {
                "fontFamily": "Nunito-SemiBold",
                "fontSize": 14,
              }
            }
          >
            Sign In
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
